Streetcode/Streetcode.XUnitTest/MediatRTests/StreetCode/Streetcode/GetAllStreetcodesHandlerTests.cs

```diff
@@ -64,9 +71,104 @@ public async Task Handle_ReturnsCorrectType()
         }
+        [Fact]
+        public async Task FindStreetCodeWithMatchTitleTest()
+        {
+            // Arrange
+            var request = new GetAllStreetcodesRequestDTO
+            {
+                Title = "Some Title"
+            };
+
+            var query = new GetAllStreetcodesQuery(request);
+
+            var (handler, repositoryWrapperMock) = SetupMockObjectsAndGetHandler(out var streetcodeRepositoryMock, out var mapperMock);
+
+            IQueryable<StreetcodeContent> streetcodes = new List<StreetcodeContent>().AsQueryable();
+
+            // Act
+            var result = await handler.Handle(query, default);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.True(result.IsSuccess);
+            Assert.NotNull(result.Value);
+
+            var response = result.Value;
+            Assert.NotNull(response.Streetcodes);
+        }
+        [Fact]
+        public async Task FindSortedStreetcodesTest()
+        {
+            // Arrange
+            var request = new GetAllStreetcodesRequestDTO
+            {
+                Sort = "-Title"
+            };
+
+            var query = new GetAllStreetcodesQuery(request);
+
+            var (handler, repositoryWrapperMock) = SetupMockObjectsAndGetHandler(out var streetcodeRepositoryMock, out var mapperMock);
+
+            var streetcodes = new List<StreetcodeContent>
+            {
+                new StreetcodeContent { Id = 1, Title = "Streetcode 1" },
+                new StreetcodeContent { Id = 2, Title = "Streetcode 2" },
+                new StreetcodeContent { Id = 3, Title = "Streetcode 3" }
+            }.AsQueryable();
+
+            // Assert
+            var result = await handler.Handle(query, default);
+            var methodInfo = typeof(GetAllStreetcodesHandler).GetMethod("FindSortedStreetcodes", BindingFlags.NonPublic | BindingFlags.Instance);
+            var parameters = new object[] { streetcodes, request.Sort };
+
+            methodInfo.Invoke(handler, parameters);
+            var sortedStreetcodes = (IQueryable<StreetcodeContent>)parameters[0];
+
+            // Act
+            var resultList = sortedStreetcodes.ToList();
+
+            Assert.Equal("Streetcode 3", resultList[0].Title);
+            Assert.Equal("Streetcode 2", resultList[1].Title);
+            Assert.Equal("Streetcode 1", resultList[2].Title);
+        }
+        [Fact]
+        public async Task FindFilteredStreetcodesTest()
+        {
+            // Arrange
+            var request = new GetAllStreetcodesRequestDTO
+            {
+                Filter = "filter:Filter"
+            };
+
+            var query = new GetAllStreetcodesQuery(request);
+
+            var (handler, repositoryWrapperMock) = SetupMockObjectsAndGetHandler(out var streetcodeRepositoryMock, out var mapperMock);
+            IQueryable<StreetcodeContent> streetcodes = new List<StreetcodeContent>().AsQueryable();
```

Comment:
why do you need explicit typing in here? consider common approach for creating variables